_posts/2016-10-16-getting-started-in-scala-on-android.md

```diff
@@ -0,0 +1,217 @@
+
+```bash
+~/.sbt/0.13/plugins/android.sbt
+```
+
+and add the following line:
+
+```bash
+addSbtPlugin("org.scala-android" % "sbt-android" % "1.7.0")
+```
+
+##Create the Android project
+Let's create a folder for our Android project:
+
+```bash
+$ mkdir my-project
+$ cd my-project
+```
+
+Now, we are going to use the SBT plugin to create a template project. First, run SBT:
+
+```bash
+$ sbt
+```
+
+Then, use the plugin to create the project.
+
+```bash
+> gen-android <package_name> <project_name>
+```
+
+For example:
+
+```bash
+> gen-android com.codurance scala_on_android
+```
+
+The project structure created looks like this:
+
+```text
+my-project/
+|-- project/
+|   |-- android.sbt
+|   |-- build.properties
+|-- src/
+|   |-- androidTest/
+|       |-- java/
+|           |-- com/
+|               |-- codurance/
+|                   |-- Junit3MainActivityTest.java
+|                   |-- Junit4MainActivityTest.java
+|   |-- main/
+|       |-- res/
+|           // Android resorces folders
+|       |-- scala/
+|           |-- com/
+|               |-- codurance/
+|                   |-- MainActivity.scala
+|       |-- AndroidManifest.xml
+|-- build.sbt
+|-- lint.xml
+```
+
+The SBT plugin creates a project structure with the minimum files needed to run an Android project, plus a setup for running instrumentation tests. Notice that the test classes generated are in Java, and the Main Activity is in Scala.
+
+The most interesting file is build.sbt. I've added some comments to explain what's the purpose of each line.
+
+```java
+// Version of the Scala runtime
+scalaVersion := "2.11.8"
+
+// Use the Android plugin
+enablePlugins(AndroidApp)
+// Add support for vector drawables
+useSupportVectors
+
+// Android version code (Same as versionCode on Gradle projects)
+versionCode := Some(1)
+// Android version name (Same as versionName on Gradle projects)
+version := "0.1-SNAPSHOT"
+
+// Instrumentation tests runner (Same as testInstrumentationRunner on Gradle projects)
+instrumentTestRunner :=
+  "android.support.test.runner.AndroidJUnitRunner"
+
+// Android platform target (Same as targetSdkVersion on Gradle projects)
+platformTarget := "android-24"
+
+// Java compile options
+javacOptions in Compile ++= "-source" :: "1.7" :: "-target" :: "1.7" :: Nil
+
+// Libraries
+libraryDependencies ++=
+  "com.android.support" % "appcompat-v7" % "24.0.0" ::
+  "com.android.support.test" % "runner" % "0.5" % "androidTest" ::
+  "com.android.support.test.espresso" % "espresso-core" % "2.2.2" % "androidTest" ::
+  Nil
+```
+
+We don't even need to use the SBT plugin to generate this template. If we prefer to craft our own minimum project, we could just create the project structure for SBT and Android manually, and add only the setup that we need.
+
+##Run the project
+You will need to have an [Android device connected](https://developer.android.com/studio/run/device.html) or an [emulator running](https://developer.android.com/studio/run/emulator.html).
```

Comment:
You will need to have a [connected Android device](https://developer.android.com/studio/run/device.html) or a [running emulator](https://developer.android.com/studio/run/emulator.html).